### PR TITLE
[Core][OTel] Tracing typing updates

### PR DESCRIPTION
**Pull request by pvaneck**
* Also updated the README to point to the right package for the OTel exporter for Azure Monitor.

### CODE REVIEW

File: sdk/core/azure-core-tracing-opentelemetry/setup.py [hunk 1]
```diff
@@ -44,7 +44,6 @@
         "Programming Language :: Python",
         "Programming Language :: Python :: 3 :: Only",
         "Programming Language :: Python :: 3",
-        "Programming Language :: Python :: 3.6",
```
Comment (kashifkhan) [inline sdk/core/azure-core-tracing-opentelemetry/setup.py:47]: while we are here, might as well add in 3.11 :)

Comment (pvaneck) [reply sdk/core/azure-core-tracing-opentelemetry/setup.py:47]: Good call. Updated :)

File: sdk/core/azure-core-tracing-opentelemetry/azure/core/tracing/ext/opentelemetry_span/__init__.py [hunk 2]
```diff
@@ -220,8 +215,7 @@ def get_trace_parent(self):
         return self.to_header()["traceparent"]
 
     @classmethod
-    def link(cls, traceparent, attributes=None):
-        # type: (str, Attributes) -> None
+    def link(cls, traceparent: str, attributes: Attributes = None) -> None:
```
Comment (kashifkhan) [inline sdk/core/azure-core-tracing-opentelemetry/azure/core/tracing/ext/opentelemetry_span/__init__.py:218]:
```suggestion
    def link(cls, traceparent: str, attributes: Optional[Attributes] = None) -> None:
```

File: sdk/core/azure-core-tracing-opentelemetry/azure/core/tracing/ext/opentelemetry_span/__init__.py [hunk 3]
```diff
@@ -231,8 +225,7 @@ def link(cls, traceparent, attributes=None):
         cls.link_from_headers({"traceparent": traceparent}, attributes)
 
     @classmethod
-    def link_from_headers(cls, headers, attributes=None):
-        # type: (Dict[str, str], Attributes) -> None
+    def link_from_headers(cls, headers: Dict[str, str], attributes: Attributes = None):
```
Comment (kashifkhan) [inline sdk/core/azure-core-tracing-opentelemetry/azure/core/tracing/ext/opentelemetry_span/__init__.py:228]:
```suggestion
    def link_from_headers(cls, headers: Dict[str, str], attributes: Optional[Attributes] = None):
```

File: sdk/core/azure-core-tracing-opentelemetry/azure/core/tracing/ext/opentelemetry_span/__init__.py (hunk 2 above)
Comment (pvaneck) [reply sdk/core/azure-core-tracing-opentelemetry/azure/core/tracing/ext/opentelemetry_span/__init__.py:218]: Typing for `attributes` follows how the `opentelemetry` package defines it where the `Optional` is part of `Attributes` (https://github.com/Azure/azure-sdk-for-python/blob/e786a17cd251327f9a5962ca825806aff9200f49/sdk/core/azure-core-tracing-opentelemetry/azure/core/tracing/ext/opentelemetry_span/__init__.py#L31)

Comment (kashifkhan) [reply sdk/core/azure-core-tracing-opentelemetry/azure/core/tracing/ext/opentelemetry_span/__init__.py:218]: ah I missed that

**Review by kashifkhan — APPROVED** [review]

**Pull request merged**